File: Assets/Scripts/UnityCMF/CCore/CcoreFactory.cs
using UnityCMF.ECore;

namespace UnityCMF.CCore {
	public interface CcoreFactory {
		C2DFieldElement CreateC2DFieldElement();
	}

	public class CcoreFactoryImpl : CcoreFactory {
		public C2DFieldElement CreateC2DFieldElement() {
			UnityCMF.ECore.EClass eClass = CcoreMeta.cINSTANCE.Package.C2DFieldElement;
			return new C2DFieldElementImpl(eClass);
		}
	}
} // UnityCMF.ccore

File: EclipseWorkspace/com.cubemonstergames.unitycmf/src/com/cubemonstergames/unitycmf/ccore/CcoreFactory.java
/**
 */
package com.cubemonstergames.unitycmf.ccore;

import org.eclipse.emf.ecore.EFactory;

/**
 * <!-- begin-user-doc -->
 * The <b>Factory</b> for the model.
 * It provides a create method for each non-abstract class of the model.
 * <!-- end-user-doc -->
 * @see com.cubemonstergames.unitycmf.ccore.CcorePackage
 * @generated
 */
public interface CcoreFactory extends EFactory {
	/**
	 * The singleton instance of the factory.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @generated
	 */
	CcoreFactory eINSTANCE = com.cubemonstergames.unitycmf.ccore.impl.CcoreFactoryImpl.init();

	/**
	 * Returns the package supported by this factory.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @return the package supported by this factory.
	 * @generated
	 */
	CcorePackage getCcorePackage();

} //CcoreFactory
